File: Week 2 - Array/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    //Calculate the average number of letters per 100 words in the text
    float avgLetters = (100 / (float)words) * (float)letters;

    //Calculate the average number of sentences per words in the text
    float avgSentences = (100 / (float)words) * (float)sentences;
    // Compute the Coleman-Liau index
    int index = round(0.0588 * avgLetters - 0.296 * avgSentences - 15.8);

    // Print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    // Return the number of letters in text
    int letters = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    // Return the number of words in text
    int words = 1;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    // Return the number of sentences in text
    int sentences = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: Week 6 - Python/sentimental-readability/readability.py
from cs50 import get_string

def main():
    input = get_string("Text: ")

    letters = count_letter(input)
    words = count_word(input)
    sentences = count_sentence(input)

    avgLetters = (100 / float(words)) * float(letters)
    avgSentences = (100 / float(words)) * float(sentences)

    index = round(0.0588 * avgLetters - 0.296 * avgSentences - 15.8)

    if index < 1:
        print("Before Grade 1")

    elif index > 16:
        print("Grade 16+")

    else:
        print(f"Grade {index}")


def count_letter(input):
    letter = 0

    for i in input:
        if i.isalpha():
            letter += 1
    return letter

def count_word(input):
    return len(input.split())

def count_sentence(input):
    sentence = 0

    for j in input:
        if j in [".", "!", "?"]:
            sentence += 1

    return sentence

main()
